web/src/app/services/HeartbeatMonitorForm.tsx

```diff
@@ -1,19 +1,32 @@
 import React from 'react'
-import p from 'prop-types'
 import Grid from '@material-ui/core/Grid'
 import TextField from '@material-ui/core/TextField'
 import { FormContainer, FormField } from '../forms'
 
-const clampTimeout = (val) => {
+function clampTimeout(val: string): number | string {
   if (!val) return ''
   const num = parseInt(val, 10)
   if (Number.isNaN(num)) return val
 
   // need to have the min be 1 here so you can type `10`
   return Math.min(Math.max(1, num), 9000)
 }
+interface HeartbeatMonitorFormProps {
+  value: {
+    name: string
+    timeoutMinutes: [number, string]
+  }
 
-export default function HeartbeatMonitorForm(props) {
+  errors: {
+    field: ['name', 'timeoutMinutes']
+    message: string
+  }
+
+  onChange: Function
```

Comment:
Define signature rather than `Function` type

Reply:
Changed to `onChange: (val: Value) => void`